internal/resources/metal/vlans/datasource_test.go

```diff
@@ -18,9 +20,9 @@ func TestAccDataSourceMetalVlan_byVxlanFacility(t *testing.T) {
 	fac := "sv15"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:          func() { testAccPreCheck(t) },
-		ExternalProviders: testExternalProviders,
-		Providers:         testAccProviders,
+		PreCheck:          func() { acceptance.TestAccPreCheckMetal(t) },
+		ExternalProviders: acceptance.TestExternalProviders,
+		ProviderFactories: acceptance.TestAccProviderFactories,
```

Comment:
I see a lot of failures in CI with error messages like this:
```

=== CONT  TestAccMetalPortCreate_hybridBonded_timeout
    resource_metal_port_acc_test.go:424: Step 1/5, expected an error with pattern, no match on: Error running pre-apply plan: exit status 1
        
        Error: Invalid resource type
        
          on terraform_plugin_test.tf line 145, in resource "equinix_metal_vlan" "test":
         145: resource "equinix_metal_vlan" "test" {
        
        The provider hashicorp/equinix does not support resource type
        "equinix_metal_vlan".
--- FAIL: TestAccMetalPortCreate_hybridBonded_timeout (0.89s)
```

Any tests that have a `metal_vlan` resource will need to be updated to use the provider factory instead of the `Provider` or `Providers` key.  I recommend picking one test to run locally and figuring out the changes needed to get just that one working, and then copy that change to other VLAN-related tests.

Reply:
Yeah looking into it